File: problems/leetcode/128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> all(nums.begin(), nums.end());
        int longest_sequence = 0;
        for (int n : all) {
            if (all.find(n - 1) == all.end()) {
                int cur_sequence = 1;
                while (all.find(n + cur_sequence) != all.end()) {
                    cur_sequence++;
                }
                longest_sequence = max(longest_sequence, cur_sequence);
            }
        }
        return longest_sequence;
    }
};

File: problems/leetcode/128.py
class Solution(object):
    def longestConsecutive(self, nums):
        numset = set(nums)
        longest = 0
        for number in numset:
            if number - 1 not in numset:
                # start of sequence
                current = 1
                while number + 1 in numset:
                    current += 1
                    number += 1
                longest = max(longest, current)
        return longest